Use new Maven plugins, add Gradle support

wsh opened the pull request.

wsh committed:
Use new Maven plugins and the most recent Java SDK.

wsh committed:
Add Gradle support; remove explicit versioning from appengine-web.xml to
reduce complexity

lesv commented on final/settings.gradle:
```diff
@@ -0,0 +1 @@
+rootProject.name = 'guestbook'
```
Probably needs newline

wsh replied:
ack

lesv commented on phase1/build.gradle:
```diff
@@ -0,0 +1,67 @@
+
+sourceCompatibility = 1.7
+targetCompatibility = 1.7
+// [END gradle]
```
newline

wsh replied:
ack

lesv commented on phase1/settings.gradle:
```diff
@@ -0,0 +1 @@
+rootProject.name = 'guestbook'
```
newline

wsh replied:
ack

lesv commented on README.md:
```diff
@@ -14,7 +14,7 @@ Please ask questions on [Stackoverflow](http://stackoverflow.com/questions/tagge
 
 1. cd final  (or phase1)
-1. `mvn clean appengine:devserver`
+1. `mvn clean appengine:run` OR `./gradlew appengineRun`
```
I prefer `gradle appengineRun` and a mention of ./gradlew.sh for folks who don't have Gradle installed.  But I'm Ok w/ this if you insist.

lesv commented on final/build.gradle:
```diff
@@ -0,0 +1,67 @@
+apply plugin: 'com.google.cloud.tools.appengine'  // App Engine tasks
+
+dependencies {
+  providedCompile 'javax.servlet:javax.servlet-api:3.1.0'
```
2.5 for standard, 3.1.0 for flex

wsh replied:
ack

lesv commented on final/build.gradle:
```diff
@@ -0,0 +1,67 @@
+    port = 8080                 // default
+  }
+
+  deploy {   // deploy configuration
```
These should go, as they are the defaults, and actually impede things in some cases.

wsh replied:
ack

lesv commented on final/build.gradle:
```diff
@@ -0,0 +1,67 @@
+
+sourceCompatibility = 1.7
+targetCompatibility = 1.7
+// [END gradle]
```
newline

wsh replied:
ack

lesv commented on final/build.gradle:
```diff
@@ -0,0 +1,67 @@
+dependencies {
+  providedCompile 'javax.servlet:javax.servlet-api:3.1.0'
+  compile 'com.google.appengine:appengine-api-1.0-sdk:+'
+  compile group: 'javax.servlet.jsp.jstl', name: 'jstl-api', version: '1.2'
```
Objectify doesn't appear as a dependency.

wsh replied:
ack, sorry!

lesv commented on phase1/build.gradle:
```diff
@@ -0,0 +1,67 @@
+apply plugin: 'com.google.cloud.tools.appengine'  // App Engine tasks
+
+dependencies {
+  providedCompile 'javax.servlet:javax.servlet-api:3.1.0'
```
2.5 for standard

wsh replied:
/facepalm thanks

lesv reviewed (commented):
PTAL - Thanks

wsh committed:
Following recommendations from
https://docs.gradle.org/current/userguide/gradle_wrapper.html about
including the gradle/ directory in source control; make lesv's suggested
changes; lock versions of App Engine SDK.

lesv reviewed (approved):
LGTM - we may not wish to merge just yet, as this is probably used in our docs somewhere.  But good to create the Veneer one from.

wsh commented:
Okay, rad. What do we need to get out of the way to merge this?
